src/routes/github/configuration/github-configuration-router.route.test.ts

```diff
@@ -0,0 +1,30 @@
+import express, { Application } from "express";
+import { GithubConfigurationRouter } from "./github-configuration-router";
+import { GithubConfigurationGet } from "./github-configuration-get";
+import { GithubConfigurationPost } from "./github-configuration-post";
+
+jest.mock("./github-configuration-get");
+jest.mock("./github-configuration-post");
+
+import supertest from "supertest";
+
+describe("GitHub Configuration Router", () => {
+	let app: Application;
+	beforeEach(() => {
+		app = express();
+		app.use("/configuration", GithubConfigurationRouter);
+	});
+	describe("Routing on GET", () => {
+		it("should hit the configuration on GET", async () => {
+			jest.mocked(GithubConfigurationGet).mockImplementation(async (_req, res) => {res.end("ok");});
+			await supertest(app).get("/configuration");
+			expect(GithubConfigurationGet).toHaveBeenCalled();
+		});
+		it("should hit the configuration on POST", async () => {
+			jest.mocked(GithubConfigurationPost).mockImplementation(async (_req, res) => {res.end("ok");});
+			await supertest(app).post("/configuration");
+			expect(GithubConfigurationPost).toHaveBeenCalled();
+		});
+	});
```

Comment:
What are you trying to test here?

Reply:
The existing `github-configuration-router.test.ts` is not testing specific to this `github-configuration-router`. It is testing the `getFrontendApp()` from the very root level. I don't think that test is appropriate to put any more unit test inside.

So in this file, I purely focus on the unit test, which is just the route.

Reply:
except, you're not testing anything except how you configured the express server _in your own test_.  That's unnecessary.  The reason we test the `getFrontendApp()` is because it contains all routes.  If someone were to change the route you're trying to test, it would actually break the test.  In this one, if you were to change the configuration route to `/configuration2` in the router, this test wouldn't break.

Reply:
1. This is test is just a "unit test", not an "integration test". 
2. This unit test test ONLY the file `github-configuration-router.ts`.
3. `github-configuration-router.ts` doesn't contains any route like `/configuration`, it just contains "/" and points to the get and post handler.

I think this test is a valid unit test. 

I agree this test is not an integration test. For example, to test the path `/configuration`, it should be somewhere else, and it will be in another PR.

Reply:
Testing the router _is_ an integration test as you're testing your config in the framework and not a single function's input/output. This test doesn't give you any extra information and if anything, gives you false confidence that everything works fine when it shouldn't.

Lean into testing the whole router without mocking and then unit test the route handler function itself for full confidence.